esmvalcore/preprocessor/_shared.py

```diff
@@ -21,54 +29,154 @@ def guess_bounds(cube, coords):
     return cube
 
 
-def get_iris_analysis_operation(operator: str) -> iris.analysis.Aggregator:
-    """Determine the iris analysis operator from a :obj:`str`.
+def get_iris_aggregator(
+    operator: str,
+    **operator_kwargs,
+) -> tuple[iris.analysis.Aggregator, dict]:
+    """Get :class:`iris.analysis.Aggregator` and keyword arguments.
 
-    Map string to functional operator.
+    Supports all available aggregators in :mod:`iris.analysis`.
 
     Parameters
     ----------
     operator:
-        A named operator.
+        A named operator that is used to search for aggregators. Will be
+        capitalized before searching for aggregators, i.e., `MEAN` **and**
+        `mean` will find :const:`iris.analysis.MEAN`.
+    **operator_kwargs:
+        Optional keyword arguments for the aggregator.
 
     Returns
     -------
-    iris.analysis.Aggregator
+    tuple[iris.analysis.Aggregator, dict]
         Object that can be used within :meth:`iris.cube.Cube.collapsed`,
         :meth:`iris.cube.Cube.aggregated_by`, or
-        :meth:`iris.cube.Cube.rolling_window`.
+        :meth:`iris.cube.Cube.rolling_window` and the corresponding keyword
+        arguments.
 
     Raises
     ------
     ValueError
-        An invalid operator is specified. Allowed options: `mean`, `median`,
-        `std_dev`, `sum`, `variance`, `min`, `max`, `rms`.
+        An invalid `operator` is specified, i.e., it is not found in
+        :mod:`iris.analysis` or the returned object is not an
+        :class:`iris.analysis.Aggregator`.
+
     """
-    operators = [
-        'mean', 'median', 'std_dev', 'sum', 'variance', 'min', 'max', 'rms',
-    ]
-    operator = operator.lower()
-    if operator not in operators:
+    cap_operator = operator.upper()
+    aggregator_kwargs = dict(operator_kwargs)
+
+    # Deprecations
+    if cap_operator == 'STD':
+        msg = (
+            f"The operator '{operator}' for computing the standard deviation "
+            f"has been deprecated in ESMValCore version 2.10.0 and is "
+            f"scheduled for removal in version 2.12.0. Please use 'std_dev' "
+            f"instead. This is an exact replacement."
+        )
+        warnings.warn(msg, ESMValCoreDeprecationWarning)
+        operator = 'std_dev'
+        cap_operator = 'STD_DEV'
+    elif re.match(r"^(P\d{1,2})(\.\d*)?$", cap_operator):
+        msg = (
+            f"Specifying percentile operators with the syntax 'pXX.YY' (here: "
+            f"'{operator}') has been deprecated in ESMValCore version 2.10.0 "
+            f"and is scheduled for removal in version 2.12.0. Please use "
+            f"`operator='percentile'` with the keyword argument "
+            "`percent='XX.YY'` instead. This is an exact replacement."
```

Comment:
give an example please, this is a bit confusing (XX.YY?)

Reply:
Added in https://github.com/ESMValGroup/ESMValCore/pull/2191/commits/4c4591640b7f9a4fa6126d9984aee2e98ac72b41